service/backend/pchain/genesis_handler.go

```diff
@@ -129,14 +129,13 @@ func (gh *gHandler) buildGenesisAllocationTx() (*txs.Tx, error) {
 
 	// TODO: this is probably not the right way to build this tx
 	// Some fields are missing that we populate in tx Builder
+	// We also do not sign/initialize the transaction since genesis outputs are referred to empty id as tx id.
```

Comment:
nit: I'd write down in the comment that this is a band-aid which breaks the invariant "all signed txs are properly signed" and that we'll look for a better solution that restores the invariant.